mantidproject/mantid
# Expand ISIS reflectometry system tests

This PR updates the reflectometry system tests to use the new workflow algorithm `ReflectometryISISLoadAndProcess` instead of the original algorithm, `ReflectometryReductionOneAuto`, which it wraps. These tests confirm that the new algorithm produces the same results as the original when called in the same way (i.e. when inputs are preprocessed before calling it). This PR also adds new system tests for doing the preprocessing steps internally in the workflow algorithm.

In detail, the changes consist of the following:
- Update the existing system tests (`INTERReductionTest` and `ISISReflectometryAutoreductionTest`) to use the new workflow algorithm. All preprocessing is still done before the algorithm is called.
- Add a new test to test the preprocessing functionality in the workflow algorithm. This is based on a cut-down version of `INTERReductionTest` using just one of the runs (plus its two transmission runs) from that test. The reference file is a cut-down version of the workspaces in the original reference file.
- Add a new test to test the slicing functionality in the workflow algorithm. Again this is a cut-down version of `INTERReductionTest`. The workspaces in the reference file differ slightly from the originals but I have confirmed that this is only due to binning differences (see the commit message for more detail).
- The tests have been refactored to use a base class to reduce duplication.
- An option has been added to use the old filter algorithms in `ReflectometrySliceEventWorkspace` - see comments below for an explanation.

**To test:**

- Code review the tests.
- Run the slicing example from [ReflectometryISISLoadAndProcess](https://docs.mantidproject.org/nightly/algorithms/ReflectometryISISLoadAndProcess-v1.html#usage).
  - Check that the slices produced in `TOF_38415_sliced` look sensible based on the given slicing properties.
  - Ensure it works with `UseNewFilterAlgorithm` set to both `True` and `False`.
  - Try setting `LogName` and `LogValueInterval` instead of `TimeInterval` to slice by log value. Note that the slices are different when using the old/new algorithm because the new algorithm uses a LogValueTolerance of 1/2 the interval width so the slices are offset by this amount. I may change the behaviour in future to be the same but this is beyond the scope of this PR.

Fixes #25881

*This does not require release notes* because **it only affects tests**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
